test/libsolidity/semanticTests/strings/empty_storage_string_1.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
+    string message;
+    event Message(string);
+    function f() external {
+        emit Message(message);
+    }
+}
```

Comment:
I'd put cases 1-5 all in a single test. Originally I wanted to suggest to give them better, more specific names but they're such minor variations that I think that just having one function going through these cases makes more sense and would be much more compact. Especially given that you're testing what happens when these values are reassigned.